src/qtils/test/outcome.hpp

```diff
@@ -249,54 +322,86 @@
       _EXPECT_OUTCOME_ERROR_2,    \
       _EXPECT_OUTCOME_ERROR_1)(__VA_ARGS__)
 
+/**
+ * @brief Expects that an expression throws ErrorException (optionally with
+ * expected error code)
+ *
+ * @param _expression_ Expression expected to throw
+ * @param _expected_code_ (optional) Expected error code
+ *
+ * @example
+ * @code
+ * EXPECT_THROW_OUTCOME(func());
+ * EXPECT_THROW_OUTCOME(func(), MyErrors::Conflict);
+ * @endcode
+ */
+#define EXPECT_THROW_OUTCOME(...) \
+  _GET_MACRO_OF_2(__VA_ARGS__,    \
+      _EXPECT_THROW_OUTCOME_2,    \
+      _EXPECT_THROW_OUTCOME_1)(__VA_ARGS__)
+
 // ------ Obsolete ------
 // TODO(xDimon): remove it at release
 
+#define _DEPRECATED_MACRO(MSG)                                \
+  ({                                                          \
+    [[deprecated(MSG)]] [[maybe_unused]] static constexpr int \
+        __deprecated_macro_warning__ = 0;                     \
+    (void)__deprecated_macro_warning__;                       \
+  })
+
 /**
  * @details This macro is obsolete. Use `ASSERT_OUTCOME_ERROR` instead.
  * @deprecated
  */
-#define ASSERT_OUTCOME_SOME_ERROR(_expression_) \
+#define ASSERT_OUTCOME_SOME_ERROR(_expression_)            \
+  _DEPRECATED_MACRO("Use `ASSERT_OUTCOME_ERROR` instead"); \
```

Comment:
May remove code without deprecation.

Reply:
I'm not sure about obsolete macros is not used anymore anywhere. I especially have added warning to explicitly recognize that.